receiver/receivercreator/config.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package receivercreator
+
+import (
+	otelconfig "github.com/open-telemetry/opentelemetry-collector/config"
+	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
+)
+
+// userConfigMap is an arbitrary map of string keys to arbitrary values as specified by the user
+type userConfigMap map[string]interface{}
+
+// subreceiverConfig is the configuration of a single subreceiver configured inside
+// receiver_creator.
+type subreceiverConfig struct {
+	// Rule is the discovery rule that when matched will create a receiver instance
+	// based on subreceiverConfig.
+	Rule string `mapstructure:"rule"`
+
+	// receiverType is set based on the configured receiver name.
+	receiverType string
+	// config is the map configured by the user in the config file. It is the contents of the map from
+	// the "config" section. The keys and values are arbitrarily configured by the user.
+	config userConfigMap
+	// fullName is the full subreceiver name (ie <receiver type>/<id>).
+	fullName string
```

Comment:
Nit: please use consistent line spacing. Either no empty lines between fields or use empty lines for all.